### DISCUSSION TITLE
Next.Js does not recognize api route

### DISCUSSION THREAD

**Discussion by Kennybabs-py** [wevm/frog-archived#384] [Q&A] [answered]
I have the frame setup in this route:

src/app/api/[[...routes]]/route.tsx

But on running localhost, I get a a blank page.
I also  get this error on running next build:

src/app/api/[[...routes]]/route.tsx
Type error: Route "src/app/api/[[...routes]]/route.tsx" does not match the required types of a Next.js Route.
  "app" is not a valid Route export field.
  
  
What am i missing?

Versions:
Next.Js : 14.1.0
Frog: 0.12.0

**Comment by Complexlity** [comment, marked as answer]
Hello, as for the first point, you'd get a blank page because frames are rendered as metatags along.
To see the page, go to [http://localhost:3000/api/dev](http://localhost:3000/api/dev)

As for the required types error, I have faced this a few times, I found that It most times had to do with imports. I'm not entirely sure what triggers it or not. One thing you could do which helped in those cases was regenerating a frog project from scratch and copying the code in while ensuring it works. That'll help you figure what triggers the error
